R/database.R

```diff
@@ -32,30 +38,38 @@ get_excluded_samples = function(tool_name = "slms-3"){
 }
 
 
-#' Get MAF-format data frame for more than one patient
+#' @title Get SSM By Patients.
 #'
-#' This function returns variants from a set of patients avoiding duplicated mutations from multiple samples from that patient(i.e. unique superset of variants).
-#' This is done either by combining the contents of individual MAF files or subsetting from a merged MAF (wraps get_ssm_by_samples)
-#' In most situations, this should never need to be run with subset_from_merge = TRUE. Instead use one of get_coding_ssm or get_ssm_by_region
+#' @description Get MAF-format data frame for more than one patient.
+#'
+#' @details This function returns variants from a set of patients avoiding duplicated mutations from multiple samples from that patient (i.e. unique superset of variants).
+#' This is done either by combining the contents of individual MAF files or subset from a merged MAF (wraps get_ssm_by_samples).
+#' In most situations, this should never need to be run with `subset_from_merge = TRUE`. Instead use one of `get_coding_ssm` or `get_ssm_by_region`
 #'
 #' @param these_patient_ids A vector of sample_id that you want results for. This is the only required argument.
-#' @param these_samples_metadata Optional metadata table
-#' @param tool_name Only supports slms-3 currently
-#' @param projection Obtain variants projected to this reference (one of grch37 or hg38)
-#' @param flavour Currently this function only supports one flavour option but this feature is meant for eventual compatibility with additional variant calling parameters/versions
-#' @param min_read_support Only returns variants with at least this many reads in t_alt_count (for cleaning up augmented MAFs)
+#' @param these_samples_metadata Optional metadata table.
```

Comment:
these_patient_ids A vector of sample_id that you want results for. 
The above argument should not be required if these_samples_metadata is provided. This is how every GAMBLR function should work. If the ids vector is missing, the function should get it from the metadata as long as the user provided it. 

these_samples_metadata Metadata table. It is expected the user has subset this to contain the rows corresponding to the patients of interest.

Reply:
Yes, 100%. I have updated the parameter descriptions, function details and examples to make this clear. The function works as advertised with the two parameters discussed.